File: web/blocks/baubles/cap.dart
// NetTango Copyright (C) Michael S. Horn, Uri Wilensky, and Corey Brady. https://github.com/NetTango/NetTango

part of NetTango;

class Cap {

  static DivElement draw(bool isTop, Block block) {
    DivElement div = new DivElement();
    div.classes.add("nt-cap");
    final blockStyle = block.getStyleClass();
    div.classes.addAll([blockStyle, "$blockStyle-color"]);

    Block.applyStyleOverrides(block, div);
    Block.maybeSetColorOverride(block.blockColor, div);

    if (isTop) {
      div.classes.add("nt-cap-top");
    } else {
      div.classes.add("nt-cap-bottom");
    }

    Block.wireDragEvents(block, div, (isOver) => block.isDragNotchOver = isOver);

    return div;
  }

}

File: src/blocks/baubles/cap.ts
// NetTango Copyright (C) Michael S. Horn, Uri Wilensky, and Corey Brady. https://github.com/NetTango/NetTango

import { Block } from "../block-instance"
import { DragCap } from "./drag-cap"

class Cap {

  static draw(isTop: Boolean, block: Block): HTMLDivElement {
    const div = DragCap.draw(isTop, block)
    Block.wireDragEvents(block, div)
    return div
  }

}

export { Cap }
